src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -105,6 +105,14 @@ public sealed class DiagnosticLogManager : AgentService, IDiagnosticLogManager
                 File.Copy(agentLogFile, destination);
             }
 
+            // Copy cloud-init log files from linux machines
+            if (PlatformUtil.RunningOnLinux)
+            {
+                executionContext.Debug("Dumping cloud-init logs.");
+                await DumpCloudInitLogs(jobStartTimeUtc, HostContext.GetDirectory(WellKnownDirectory.Diag));
+                executionContext.Debug("Dumping cloud-init logs is completed.");
```

Comment:
Will it be really completed if error happens? Probably some re-phrasing would make sense as well